test/functional/api/buffer_spec.lua

```diff
@@ -35,8 +35,42 @@ describe('api/buf', function()
       -- There's always at least one line
       eq(1, curbuf_depr('line_count'))
     end)
-  end)
 
+    it('line_count has defined behaviour for unloaded buffers', function()
+      -- we'll need to know our bufnr for when it gets unloaded
+      local bufnr = request('nvim_buf_get_number', 0)
+      -- replace the buffer contents with these three lines
+      request('nvim_buf_set_lines', bufnr, 0, -1, 1, {"line1", "line2", "line3", "line4"})
+      -- check the line count is correct
+      eq(4, request('buffer_line_count', bufnr))
+      eq(4, request('nvim_buf_line_count', bufnr))
+      -- force unload the buffer (this will discard changes)
+      request('nvim_command','new')
+      request('nvim_command', 'bunload! '..bufnr)
+      -- line count for an unloaded buffer should always be 0
+      eq(0, request('buffer_line_count', bufnr))
```

Comment:
obsolete functions (or even just an obsolete name in this case) need not be tested in new tests, unless there is some edge case that affects the obsolete function or something (which is not here, this this just an alias)